include/stlink/stlinkusb.h

```diff
@@ -0,0 +1,38 @@
+#ifndef STLINKUSB_H
+#define STLINKUSB_H
+
+#include <libusb.h>
+
+/*
+
+ libusb ver  | LIBUSB_API_VERSION
+-------------+--------------------
+   v1.0.13   |    0x01000100
+   v1.0.14   |    0x010000FF
+   v1.0.15   |    0x01000101
+   v1.0.16   |    0x01000102       < MINIMAL_API_VERSION
+   v1.0.17   |    0x01000102
+   v1.0.18   |    0x01000102
+   v1.0.19   |    0x01000103
+   v1.0.20   |    0x01000104
+   v1.0.21   |    0x01000105
+   v1.0.22   |    0x01000106
+   v1.0.23   |    0x01000107
+
+*/
+
+#if !defined ( LIBUSBX_API_VERSION )
+#if defined (__FreeBSD__)
+#define LIBUSBX_API_VVERSION LIBUSB_API_VERSION
+#elif !defined (LIBUSB_API_VERSION)
+#error unsupported libusb version
+#endif
+#endif
+
+#define MINIMAL_API_VERSION 0x01000102
+
+#if ( LIBUSB_API_VERSION < MINIMAL_API_VERSION )
+#error unsupported libusb version
+#endif
+

```

Comment:
Check typo to avoid bug: LIBUSBX_API_**V**VERSION --> LIBUSB_API_VERSION

... but we may simplify this (suggestion):

/* for all systems */
#if defined ( LIBUSBX_API_VERSION )
#error unsupported libusb version
#endif

/* FreeBSD */
// libusb is integrated in FreeBSD, only ensure 1.0.26 or later codebase...
#if defined (__ FreeBSD __)
#define MIN_LIBUSB_API_VERSION 0x01000102 
#if ( LIBUSB_API_VERSION < MIN_LIBUSB_API_VERSION ) 
#error unsupported libusb version
#endif
#endif

/* Linux-Distributions */
// install-package used --> version check via cmake compilation routine

/* macOS */
// TBD

/* Windows */
// external install-package used --> version check and fetch via cmake compilation routine if necessary

Reply:
See also #733 regarding LIBUSBX_API_VERSION. We don't need it anymore at all not even for FreeBSD.
By now (#897) we would solely define libusb versions that don't have it in their codebase anymore.

Reply:
this is incorrect
```c
#if defined ( LIBUSBX_API_VERSION )
#error unsupported libusb version
#endif
```
because it really defined for compatibility.

So, I prefer stay this without changes at this time.

Reply:
Yes, you are right I have convinced myself which 1.0.18 as a reference.
If we don't keep `LIBUSBX_API_VERSION` until this define finally vanishes from `libusb.h`, they will all fail to install.
Good that we had a separate look at it.

Reply:
We may do it like this then:

```
/* libusb is integrated in FreeBSD */
#if defined (__FreeBSD__)

    /* Verify API-Type */
    #if !defined (LIBUSBX_API_VERSION)
        // (still) defined in libusb-codebase for compatibility (to be deprecated in the future)
        #define LIBUSBX_API_VERSION LIBUSB_API_VERSION
    #elif !defined (LIBUSB_API_VERSION)
        #error unsupported libusb version // ... we find to have a libusbx-codebase
    #endif

    /* Verify API-Version */
    #define MIN_LIBUSB_API_VERSION 0x01000102
    #if ( LIBUSB_API_VERSION < MIN_LIBUSB_API_VERSION )
        #error unsupported libusb version // ... we find libusb-codebase to be older than 1.0.16
    #endif
#endif
```

/* Linux-Distributions */
// install-package used --> version check via cmake compilation routine

/* macOS */
// TBD

/* Windows */
// external install-package used --> version check and fetch via cmake compilation routine if necessary

Reply:
I think current checks are enough at this moment.
If issues will be founded we fix them.

Reply:
Discussed this with a conclusion resulting in commit db6e93e52aa86add8ff0453ca2d26ca13310fa8e.